src/Storages/MergeTree/MergeTreeReaderStream.cpp

```diff
@@ -195,18 +195,25 @@ void MergeTreeReaderStream::adjustForRange(MarkRange range)
     auto [right_offset, mark_range_bytes] = getRightOffsetAndBytesRange(range.begin, range.end);
     if (!right_offset)
     {
+        if (last_right_offset && !last_right_offset.value())
+            return;
+
+        last_right_offset = 0; // Zero value means the end of file.
```

Comment:
why we need this?

Reply:
Same reason as I wrote on your first comment.